libraries/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -178,17 +196,29 @@ export class ProjectChangeAnalyzer {
     return projectState;
   }
 
+  /**
+   * Given a specific project and a list of changed files for that project, apply project-specific
+   * "ignore globs" settings, as well as an optional filtering function, returning a new list of
+   * changed files that have passed through the filter.
+   */
   public async _filterProjectDataAsync<T>(
     project: RushConfigurationProject,
     unfilteredProjectData: Map<string, T>,
     rootDir: string,
-    terminal: ITerminal
+    terminal: ITerminal,
+    filterByProject?: ProjectFilterFactory
```

Comment:
This should just be the filter itself (which is synchronous), which the caller is expected to provide in its entirety (i.e. including any behavior from `RushProjectConfiguration`). The motivation is to make the filtering call be completely synchronous and not perform any file system calls.